### PR TITLE
Main: Add a generic tracing facility

### PR DESCRIPTION
**Pull request by pragmaware**
Add a generic tracing facility.
Rewrite the CXX_ specific trace macros.

### CODE REVIEW

File: main/parse.c [hunk 1]
```diff
@@ -2434,6 +2438,7 @@ extern bool parseFileWithMio (const char *const fileName, MIO *mio)
 	if (Option.printLanguage)
 	{
 		printGuessedParser (fileName, language);
+		TRACE_LEAVE();
```
Comment (masatake) [inline main/parse.c]: It seems that there is no associated TRACE_ENTER for this TRACE_LEAVE. It is o.k.?

What I got:

```
% ./ctags --print-language main/main.c
[>> parseFile][at 0] Parsing file main/main.c
main/main.c: C
[<< parseFileWithMio][at 0] 
[<< parseFile][at 0] 
```

**Comment by masatake** [comment]
Could you remove unnecessary whitespaces reported by git show:

whitespaces

File: main/parse.c (hunk 1 above)
Comment (pragmaware) [inline main/parse.c]: Right... thnx.

**Comment by masatake** [comment]
Thank you!

**Pull request merged**